### PR TITLE
[C5][environment] Creating new Rest API component ConfigurationsAPI

### PR DESCRIPTION
**Pull request by RenukaFernando**
### Proposed changes in this pull request
- Create new component `org.wso2.carbon.apimgt.rest.api.configurations`.
- Create new Rest API `ConfigurationsAPI` to read any configurations.
- Add `environments` resource to `ConfigurationAPI`.

This will allow following configuration segment in deployment.yaml
```yaml
wso2.carbon.apimgt.environments:
  environmentName: Default
  environments:
  - host: dev.sample.com:9292
    loginTokenPath: /login/token
    label: Development

  - host: prod.sample.com:9292
    loginTokenPath: /login/token
    label: Production
``` 
### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/src/main/java/org/wso2/carbon/apimgt/rest/api/configurations/ConfigurationService.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.configurations;
+
+import org.wso2.carbon.apimgt.rest.api.configurations.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.rest.api.configurations.models.EnvironmentConfigurations;
+
+/**
+ * ervice class for get configurations
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/src/main/java/org/wso2/carbon/apimgt/rest/api/configurations/ConfigurationService.java]: Spelling mistake.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/pom.xml [hunk 2]
```diff
@@ -0,0 +1,211 @@
+<?xml version="1.0"?>
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/pom.xml]: Shall we add the License header to this pom as well ?

File: features/apimgt/org.wso2.carbon.apimgt.rest.api.configurations.feature/pom.xml [hunk 3]
```diff
@@ -0,0 +1,121 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2016, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.rest.api.configurations.feature/pom.xml]: Incorrect year in the license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/src/main/java/org/wso2/carbon/apimgt/rest/api/configurations/ConfigurationService.java (hunk 1 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/src/main/java/org/wso2/carbon/apimgt/rest/api/configurations/ConfigurationService.java]: Fixed issue with https://github.com/RenukaFernando/carbon-apimgt/commit/b587c62656dbb084d324e343f09c7f9f8ab1672a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/pom.xml (hunk 2 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/pom.xml]: Fixed issue with https://github.com/RenukaFernando/carbon-apimgt/commit/b587c62656dbb084d324e343f09c7f9f8ab1672a

File: features/apimgt/org.wso2.carbon.apimgt.rest.api.configurations.feature/pom.xml (hunk 3 above)
Comment (RenukaFernando) [reply features/apimgt/org.wso2.carbon.apimgt.rest.api.configurations.feature/pom.xml]: Fixed issue with https://github.com/RenukaFernando/carbon-apimgt/commit/b587c62656dbb084d324e343f09c7f9f8ab1672a

**Pull request merged**